controllers/ptpinterface_controller.go

```diff
@@ -674,3 +691,84 @@ func (r *PtpInterfaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	config, ok := annotation[cloudManager.RestoreInProgress]
+	if ok {
+		restoreStatus := &cloudManager.RestoreStatus{}
+		err := json.Unmarshal([]byte(config), &restoreStatus)
+		if err == nil {
+			if restoreStatus.InSync != nil {
+				instance.Status.InSync = *restoreStatus.InSync
+			}
+			if restoreStatus.DeploymentScope != nil {
+				if strings.ToLower(*restoreStatus.DeploymentScope) == "bootstrap" ||
+					strings.ToLower(*restoreStatus.DeploymentScope) == "principal" {
+					instance.Status.DeploymentScope = *restoreStatus.DeploymentScope
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid deploymentScope value '%s', scope will be set to boostrap", *restoreStatus.DeploymentScope)
+					instance.Status.DeploymentScope = "bootstrap"
+				}
+			} else {
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "The 'deploymentScope' is not specified, assuming 'bootstrap' scope.")
+				instance.Status.DeploymentScope = "bootstrap"
+			}
+			if restoreStatus.StrategyRequired != nil {
+				if *restoreStatus.StrategyRequired == "not_required" ||
+					*restoreStatus.StrategyRequired == "lock_required" ||
+					*restoreStatus.StrategyRequired == "unlock_required" {
+					instance.Status.StrategyRequired = *restoreStatus.StrategyRequired
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid strategyRequired value '%s', it will be set to 'not_required'", *restoreStatus.StrategyRequired)
+					instance.Status.StrategyRequired = "not_required"
+				}
+			} else {
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "The 'strategyRequired' is not specified, assuming 'not_required'.")
+				instance.Status.StrategyRequired = "not_required"
+			}
+			instance.Status.Reconciled = true
+			instance.Status.ObservedGeneration = instance.ObjectMeta.Generation
+			err = r.Client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				log_err_msg := fmt.Sprintf(
+					"Failed to update ptp interface status while restoring '%s' resource. Error: %s",
+					instance.Name,
+					err)
+				return common.NewResourceStatusDependency(log_err_msg)
+			} else {
+				StatusUpdate := fmt.Sprintf("Status updated for PtpInterface resource '%s' during restore with following values: Reconciled=%t InSync=%t DeploymentScope=%s",
+					instance.Name, instance.Status.Reconciled, instance.Status.InSync, instance.Status.DeploymentScope)
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, StatusUpdate)
+
+			}
+		} else {
+			r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Failed to unmarshal '%s'", err)
+		}
+	}
+	return nil
+}
+
+// Clear annotation RestoreInProgress
+func (r *PtpInterfaceReconciler) ClearAnnotation(instance *starlingxv1.PtpInterface) error {
```

Comment:
1 Need to rename it to "ClearRestoreInProgress", it is not clearing all the annotations
2 Important: there are multiple clearAnnotation in all the controllers, do we need to wait for all the steps done before the actual clear annotation?

Reply:
1 -> acknowledged
2 -> restore-in-progress annotation is the first to be processed and hence the first to be cleared as well, no dependency on other annotations as such